src/main/java/org/candlepin/subscriptions/tally/InventoryAccountUsageCollector.java

```diff
@@ -247,24 +247,34 @@ protected void reconcileHbiSystemWithSwatchSystem(
       Set<String> applicableProducts,
       List<Host> hosts) {
     log.debug(
-        "Reconciling HBI inventoryId={} & swatch inventoryId={}",
+        "Reconciling HBI inventoryId={} & swatch inventoryId={} & swatch instanceId={}",
         Optional.ofNullable(hbiSystem).map(InventoryHostFacts::getInventoryId),
-        Optional.ofNullable(swatchSystem).map(Host::getInventoryId));
+        Optional.ofNullable(swatchSystem).map(Host::getInventoryId),
+        Optional.ofNullable(swatchSystem).map(Host::getInstanceId));
     if (hbiSystem == null && swatchSystem == null) {
       log.debug("Unexpected, both HBI & Swatch system records are empty");
     } else if (hbiSystem == null) {
-      log.info("Deleting system w/ inventoryId={}", swatchSystem.getInventoryId());
+      log.info(
+          "Deleting system w/ inventoryId={} and instanceId={}",
+          swatchSystem.getInventoryId(),
+          swatchSystem.getInstanceId());
```

Comment:
We'll need to add an additional check here: if we have seen a system in metering data (instance type "RHEL System"), but there's no HBI record, it should be kept/skipped.

Reply:
This sounds to me like an unrelated task. Let's clarify this point internally.

Reply:
given `isMetered` checks, no longer relevant

Reply:
do you mean we don't need to check the instance type because we're already checking the isMetered flag?